design/authenticators_health_api.md

```diff
@@ -0,0 +1,233 @@
+Let's say the Conjur Operator configured the OIDC Authenticator and forgot to whitelist 
+the authenticator in the CONJUR_AUTHENTICATORS variable. Without the Helthcheck API, 
+the first time the user will encounter the error will be in an authentication request. 
+A user `alice` will try to authenticate with OIDC and will get a 401 Unauthorized 
+response without any description on what happened (the answer will be in the logs which 
+are probably unaccessible to her) even if the actual request is ok. The main problem 
+here is that the Unauthorized response has nothing to do with the actual request, as 
+the problem is not in it. The problem is in the configuration which was made by another 
+person in another time. With the Healthcheck API the person who does the configuration will 
+get an immediate feedback on it, and the users that are *using* the authenticators will not 
+be unauthorized due to configuration errors.
+
+### Current Mechanism
+
+The current mechanism for understanding that there is a problem in the Authenticator configuration is as follows:
+
+1. The Operator configures the Authenticator according to the docs
+1. A user tries to authenticate\
+    1. this may be the Operator himself imitating a user to verify the authenticator works in the dev env
+1. The user gets a 401 Unauthorized response
+    1. There are other response codes possible but configuration issues will be 401
+1. The user reports to the operator that there is an issue
+    1. This one is tricky as the user might not know who to report this to.
+    1. Again, this may be the operator himself
+1. The operator looks in the logs (according to the docs) and finds the issue
+1. The operator fixes the issue
+
+This can be iterative as there might be several issues. It is possible to assume that in the case 
+above, the operator will solve this with the user and will ask him to try to authenticate again and 
+if it's still Unauthorized will solve the configuration issues one-by-one. 
+
+### Process Logic
+
+1. The Conjur operator logs into Conjur
+1. The Conjur operator configures the Authenticators according to the docs
+    1. We might have several authenticators configured
+1. The operator runs the Healthcheck request: `GET /authenticators/health`
+1. The operator gets a response:
+    1. In case all the authenticators are healthy, a response with code 200 and the following body:
+        
+        ```
+        {
+           “authenticators”:
+           {
+              "authn-1": "ok",
+              "authn-2": "ok"
+           },
+           "ok": true
+        }
+        ```
+    1. In case some of the authenticators aren't healthy, a response with code 500 (more info [here](authenticators_health_api.md#response-code-for-unhealthy-authenticators)) and the following body:
+
+        ```
+        {
+           “authenticators”:
+           {
+              "authn-1": "ok",
+              "authn-2": "<configuration error>",
+           },
+           "ok": true
```

Comment:
should this be `"status": "error"` as above? can we copy the response format from the previous design doc here as well?

Reply:
oh this should actually be "false", thanks!

Reply:
and same as above - using `"ok": false` instead of `"status": "error"` to maintain ux of general health-check